src/test/java/org/jabref/logic/importer/fetcher/GrobidCitationFetcherTest.java

```diff
@@ -0,0 +1,112 @@
+    static String example1AsBibtex = "@article{-1,\n" +
+            "author\t=\t\"T Derwing and M Rossiter and M Munro\",\n" +
+            "title\t=\t\"Teaching native speakers to listen to foreign-accented speech\",\n" +
+            "journal\t=\t\"Journal of Multilingual and Multicultural Development\",\n" +
+            "year\t=\t\"2002\",\n" +
+            "pages\t=\t\"245--259\",\n" +
+            "volume\t=\t\"23\",\n" +
+            "number\t=\t\"4\"\n" +
+            "}";
+    static String example2 = "Thomas, H. K. (2004). Training strategies for improving listeners' comprehension of foreign-accented speech (Doctoral dissertation). University of Colorado, Boulder.";
+    static String example2AsBibtex = "@misc{-1,\n" +
+            "author\t=\t\"H Thomas\",\n" +
+            "title\t=\t\"Training strategies for improving listeners' comprehension of foreign-accented speech (Doctoral dissertation)\",\n"
+            +
+            "year\t=\t\"2004\",\n" +
+            "address\t=\t\"Boulder\"\n" +
+            "}";
+    static String example3 = "Turk, J., Graham, P., & Verhulst, F. (2007). Child and adolescent psychiatry : A developmental approach. Oxford, England: Oxford University Press.";
+    static String example3AsBibtex = "@misc{-1,\n" +
+            "author\t=\t\"J Turk and P Graham and F Verhulst\",\n" +
+            "title\t=\t\"Child and adolescent psychiatry : A developmental approach\",\n" +
+            "publisher\t=\t\"Oxford University Press\",\n" +
+            "year\t=\t\"2007\",\n" +
+            "address\t=\t\"Oxford, England\"\n" +
+            "}";
+    static String example4 = "Carr, I., & Kidner, R. (2003). Statutes and conventions on international trade law (4th ed.). London, England: Cavendish.";
+    static String example4AsBibtex = "@article{-1,\n" +
+            "author\t=\t\"I Carr and R Kidner\",\n" +
+            "booktitle\t=\t\"Statutes and conventions on international trade law\",\n" +
+            "publisher\t=\t\"Cavendish\",\n" +
+            "year\t=\t\"2003\",\n" +
+            "address\t=\t\"London, England\"\n" +
+            "}";
+    static String invalidInput1 = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx________________________________";
+    static String invalidInput2 = "¦@#¦@#¦@#¦@#¦@#¦@#¦@°#¦@¦°¦@°";
+    static BibEntry example1AsBibEntry;
+    static BibEntry example2AsBibEntry;
+    static BibEntry example3AsBibEntry;
+    static BibEntry example4AsBibEntry;
+
+    @BeforeAll
+    public static void setup() throws ParseException {
+        importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
+        fileUpdateMonitor = new DummyFileUpdateMonitor();
+        grobidCitationFetcher = new GrobidCitationFetcher(importFormatPreferences);
+        example1AsBibEntry = BibtexParser.singleFromString(example1AsBibtex, importFormatPreferences, fileUpdateMonitor).get();
```

Comment:
Please don't parse the entries from string but create them by hand using `new BibEntry().withField(..., ...)`.

Reply:
Solved here: https://github.com/JabRef/jabref/pull/5614/commits/9af46abac88d8f45beb81150619736ca60f3a4c3